### ISSUE TITLE
Nested timelines (branches) may loose data

### ISSUE THREAD

**Issue by ololobus** [zenithdb/zenith#1371]
We have a basic branching e2e test in the console:
https://github.com/zenithdb/console/blob/2e1ffccb242e2c3f157029c5a7d6a5dfc97f5e69/tests_e2e/tests/test_cluster.py#L385-L416

And it [fails from time to time](https://github.com/zenithdb/console/runs/5573659053?check_suite_focus=true#step:26:261) with:
```
resp       = Response(status_code=200, content=b'{"success":true,"response":[{"query":"select count(*) from t;","error":"ERROR: relation \\"t\\" does not exist (SQLSTATE 42P01)","explain_data":null,"data":{"fields":null,"rows":null,"truncated":false}}]}', 
```
Which means that freshly created nested timeline doesn't see old data. Feel free to close if it's a known problem.

**Comment by LizardWizzard** [comment]
My guess is that wal for the inserted data didn't reach the pageserver at the time of create branch call. So branch was possibly created at an lsn that is earlier than that of the inserted data.

We may want to consider taking the lsn for creating a branch from compute/safekeeper to be at the real tip of the branch. When pageserver is asked to create a branch at an lsn that is not currently ingested it waits until needed wal arrives and then performs the branching operation (maybe it is better to error out immediately instead of waiting)

**Comment by kelvich** [comment]
We could fix that by asking the latest lsn from safekeepers before the branch creation, so that branch creation always happen with some lsn. 
Each safekeeper have only partial view of the world, so we have to talk to the majority of them to understand the proper lsn.

@arssher @petuhovskiy what could be the algorithm here for the console? I think we cannot do that in entirely correct way without performing an election, but can we come up with reasonable heuristic (worst case is that branch creation will fail)? I guess asking for `sort(last_received_lsn)[1]` will do the trick? Code should also work with only two safekeepers alive.

**Comment by stepashka** [comment]
is this a symptom of neondatabase/neon#2579 ?

**Comment by kelvich** [comment]
> is this a symptom of https://github.com/neondatabase/neon/issues/2579 ?

Yes, both issues are about the same problem. In test we've fixed that race by 10 seconds sleep. I'd keep this issue over https://github.com/neondatabase/neon/issues/2579

**Comment by ololobus** [comment]
> I'd use commit_lsn instead

Also thought about using it instead. Going to do LSN validation in the https://github.com/neondatabase/cloud/issues/2637 and probably this 'HEAD through the last known commit LSN' today-tomorrow on the console side

**Comment by stepashka** [comment]
are you working on a solutio for this, @ololobus
